### DISCUSSION TITLE
Distinguish player provider type in player selection

### DISCUSSION THREAD

**Discussion by galligan** [music-assistant/support#2696] [Q&A] [answered]
Currently I have a mix of speaker types in my house: Sonos, HomePod (AirPlay), and Devialet (DNLA), and have them configured in Music Assistant. However, because the Sonos speakers are also configured to show up as AirPlay speakers, there is duplication in the player selection. See image here:

![IMG_0059](https://github.com/user-attachments/assets/3a01e79b-245a-49ed-9c9e-e13d65b2e3b3)

The simpler solution would be to distinguish the player type through a label or symbol, indicating what type of player it is. This way I could distinguish between my Sonos and AirPlay easily.

The more complex, but more user-friendly and powerful solution, would to have deduplication and intelligent player choice powering this feature. This way, if I want to play to my Kitchen Sonos, by default it would use the Sonos route if the audio source is compatible. But if I already have a HomePod playing, and I wanted to group a Sonos speaker with it, then it would default to AirPlay as the protocol.

**Comment by OzGav** [comment, marked as answer]
https://music-assistant.io/ui/?h=icon#player-list
